spec/integration/users_spec.rb

```diff
@@ -0,0 +1,103 @@
+  #   end
+  #   let(:users) { User.all }
+  #   it 'shows the right content' do
+  #     visit root_path
+  #     users.each do |user|
+  #       expect(page).to have_content(user.name)
+  #     end
+  #   end
+
+  #   it 'shows the correct photo' do
+  #     visit root_path
+  #     users.each do |user|
+  #       expect(page).to have_css("img[src*='#{user.photo}']")
+  #     end
+  #   end
+
+  #   it 'shows posts counter' do
+  #     visit root_path
+  #     users.each do |user|
+  #       expect(page).to have_content("Number of posts: #{user.posts.count}")
+  #     end
+  #   end
+  # end
+
+  # describe 'users show' do
+  #   let(:user) { User.new(name: 'Tango', photo: 'http://myphoto') }
+  #   let(:post) { Post.new(author: user, title: 'blablabla', text: 'blablabla') }
+
+  #   before(:each) do
+  #     user.save
+  #     post.save
+  #   end
+
+  #   it "shows the user's profile picture." do
+  #     visit user_path(user.id)
+  #     expect(page).to have_css("img[src*='#{user.photo}']")
+  #   end
+
+  #   # I can see the user's username.
+  #   it "shows the user's username." do
+  #     visit user_path(user.id)
+  #     expect(page).to have_content(user.name)
+  #   end
+
+  #   it 'shows the number of posts the user has written.' do
+  #     visit user_path(user.id)
+  #     expect(page).to have_content("Number of posts: #{user.posts.count}")
+  #   end
+
+  #   it "shows the user's bio." do
+  #     visit user_path(user.id)
+  #     expect(page).to have_content(user.bio)
+  #   end
+
+  #   # I can see the user's first 3 posts.
+  #   it "shows the user's first 3 posts." do
+  #     visit user_path(user.id)
+  #     user.recent_posts.each do |post|
+  #       expect(page).to have_content(post.title)
+  #     end
+  #   end
+  #   # I can see a button that lets me view all of a user's posts.
+  #   it "shows a button that lets me view all of a user's posts." do
+  #     visit user_path(user.id)
+  #     expect(page).to have_button('Show All Posts')
+  #   end
+  # end
+
+  describe 'checks the links' do
+    before(:each) do
+      user = User.create(name: 'Tango', photo: 'http://myphoto')
+      Post.create(author: user, title: 'blablabla', text: 'blablabla')
+    end
+    let(:user) { User.first }
+    let(:post) { user.posts[0] }
+    it 'the user is redirected to the correct url' do
+      visit root_path
+
+      click_link user.name
+      expect(page).to have_current_path(user_path(user.id))
+    end
+
+    it "redirects me to post's show page When I click a user's post." do
+      visit user_posts_path(user.id)
+
+      click_link post.title
+      expect(page).to have_current_path(user_post_path(user_id: user.id, id: post.id))
+    end
+    it "redirects me to the user's post's index page When I click to see all posts" do
+      visit user_path(user.id)
+
+      click_link 'Show All Posts'
+      expect(page).to have_current_path(user_posts_path(user_id: user.id))
+    end
```

Comment:
- [ ] Great job so far. However, there're few things you still need to do some test for your milestone to complete its requirements. All the underline points on the screenshot are the tests you're yet to write for this page.

### User index
![user-index](https://github.com/Muhammad0602/Blog-app/assets/26597432/35f10ea3-7d01-412b-8f98-45b848212151)

- [ ] It is also required that spec file should only test User index page. It can be only one spec testing all off the things below or more, ideally one for the asserts and one for the redirect.
   -  ### One file for the user index page
       -  I can see the username of all other users.
       -  I can see the profile picture for each user.
       -  I can see the number of posts each user has written.
       -  When I click on a user, I am redirected to that user's 
           show page.
    -  ### Another file for the user show page
        -   can see the user's profile picture.
        -   I can see the user's username.
        -   I can see the number of posts the user has written.
        -   I can see the user's bio.
        -  I can see the user's first 3 posts.
        -  I can see a button that lets me view all of a user's 
            posts.
        -   When I click a user's post, it redirects me to that 
            post's show page.
        -  When I click to see all posts, it redirects me to the 
           user's post's index page.

This format should be applied for the rest of the tests.

Reply:
Hello @Qoosim. thanks for the feedback. The commented code is indeed the code for these underlined tests, we forgot to uncomment them. 
Can you please specify where it is stated to have different files for the tests in the requirements? As far as I know, it is stated to think about it and come up with something that makes sense, but it doesn't say to create different files for index and show pages.